numerical/area/bound.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+from abc import ABCMeta, abstractmethod
+import matplotlib.pyplot as plt
+
+
+class _BaseBoundary(metaclass=ABCMeta):
+    @abstractmethod
+    def plot(self, save_path=None):
+        """ Visualizes boundary. """
+
+    def _set_described_rect(self):
+        """ Set n-dim rectangular bound which describes initial bound. """
+
+
+class _BoundaryIterable(metaclass=ABCMeta):
+    """ The class represents iterator interface for boundaries. """
+    def __init__(self, data: list):
+        self._data = data
+        self._bounds_count = len(data)
+
+    def __iter__(self):
+        self._bound_id = -1
+        return self
+
+    def __next__(self):
+        if self._bound_id == self._bounds_count - 1:
+            raise StopIteration
+        self._bound_id += 1
+        return self._data[self._bound_id]
+
+    def __getitem__(self, item):
+        return self._data[item]
+
+    def __repr__(self):
+        return "<[" + ", ".join([bnd.__repr__() for bnd in self._data]) + "]>"
+
+
+class _LineBoundary(_BaseBoundary):
+    """ The class represents start and end of 1-dim line. """
+    def __init__(self, start, end):
+        if end <= start:
```

Comment:
bad comparison for float, use abs value from difference

Reply:
OK

Reply:
https://stackoverflow.com/a/33024979
My concern is only about `==` operation. Use `math.isclose`